GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -436,10 +436,11 @@ public static ArgumentString RebaseCmd(string branch, bool interactive, bool pre
 
             return new GitArgumentBuilder("rebase")
             {
-                { interactive, "-i" },
-                { interactive && autosquash, "--autosquash" },
-                { interactive && !autosquash, "--no-autosquash" },
-                { preserveMerges, GitVersion.Current.SupportRebaseMerges ? "--rebase-merges" : "--preserve-merges" },
+                { ignoreDate, "--ignore-date" },
+                { interactive && !ignoreDate, "-i" },
+                { interactive && autosquash && !ignoreDate, "--autosquash" },
+                { interactive && !autosquash && !ignoreDate, "--no-autosquash" },
+                { preserveMerges && !ignoreDate, GitVersion.Current.SupportRebaseMerges ? "--rebase-merges" : "--preserve-merges" },
```

Comment:
nitpick: I'd put `!ignoredate' in front in order the emphasize the incompatibility. Thoughts?

Reply:
I don't have a strong preference.

Maybe it worth structuring differently, e.g.:
```cs
            var builder = new GitArgumentBuilder("rebase");
            if (ignoreDate)
            {
                builder.Add("--ignore-date");
            }
            else
            {
                builder.AddRange({ ... });
            }

            builder.AddRange({ /* common */ });
```

Reply:
Yes, this will be clearer.

Reply:
I agree. As much as I appreciate the initializer, I think it's fairly unclear to use with this kind of logic. I think going back to a standard selection statement is the way to go, especially since I am also planning on adding another checkbox with the same types of incompatibilities.